test/lib/auth.js

```diff
@@ -63,6 +69,179 @@ describe('Auth lib', function () {
+	    auth.checkOrganizations(req, {}, function (){
+		done();
+	    });
+	});
+	
+	it ('should continue with middleware processing if createOrganization call fails', function (done) {
+
+	    var getOrganizations = function (callback){
+		return callback(null,
+				{status: 200,
+				 body: [{id: '123456789',
+					 name: 'patty',
+					 href: 'www.exampleuri.com/orgs/patty'},
+					{id: '987654321',
+					 name: 'MntyPythn',
+					 href: 'www.exampleuri.com/orgs/MntyPythn'}]
+				})
+	    };
+	    var createOrganization = function (content, callback){
+		return callback({status: 500, message: 'An error occurred during request'})
+	    };
+	    var auth = getAuthLib(strategy, {}, null, getOrganizations, createOrganization);
+	    
+	    auth.getCache()['token'] = { orgProcessed : false, orgProcessing: true};
+	    auth.checkOrganizations(req, {}, function () {
+		done();
+	    });
+	});
+
+	it('should continue with middleware processing if updateIndividual call fails', function (done) {
+	    var getOrganizations = function (callback){
+		return callback(null,
+				{status: 200,
+				 body: [{id: '123456789',
+					 name: 'patty',
+					 href: 'www.exampleuri.com/orgs/patty'},
+					{id: '987654321',
+					 name: 'MntyPythn',
+					 href: 'www.exampleuri.com/orgs/MntyPythn'}]
+				})
+	    };
+	    var createOrganization = function (content, callback){
+		return callback(null,
+				{id: '111555999',
+				 tradingName: 'AmaneceQueNoEsPoco',
+				 href: 'www.exampleuri.com/orgs/AmaneceQueNoEsPoco'})
+	    };
+	    var updateIndividual = function (id, roles, callback){
+		return callback({status: 500, message: 'An error occurred during request'})
+	    };
+	    var auth = getAuthLib(strategy, {}, null, getOrganizations, createOrganization, updateIndividual);
+
+	    auth.getCache()['token'] = { orgProcessed : false, orgProcessing: true};
+	    auth.checkOrganizations(req, {}, function () {
+		done();
+	    });
+	    
+	});
+	
+	it('should continue with middleware processing after updating partyAPI backend data', function (done) {
+	    var getOrganizations = function (callback){
+		return callback(null,
+				{status: 200,
+				 body: [{id: '123456789',
+					 name: 'patty',
+					 href: 'www.exampleuri.com/orgs/patty'},
+					{id: '987654321',
+					 name: 'MntyPythn',
+					 href: 'www.exampleuri.com/orgs/MntyPythn'}]
+				})
+	    };
+	    var createOrganization = function (content, callback){
+		return callback(null,
+				{id: '111555999',
+				 tradingName: 'AmaneceQueNoEsPoco',
+				 href: 'www.exampleuri.com/orgs/AmaneceQueNoEsPoco'})
+	    };
+	    var updateIndividual = function (id, roles, callback){
+		return callback(null, {status: 200})
+	    };
+	    var auth = getAuthLib(strategy, {}, null, getOrganizations, createOrganization, updateIndividual);
+
+	    auth.getCache()['token'] = { orgProcessed : false, orgProcessing: true};
+	    auth.checkOrganizations(req, {}, function () {
+		done();
```

Comment:
Call validation. In this test you can see that without validating the call you cannot distinguish between this test and the previous one